x-pack/plugins/canvas/canvas_plugin_src/renderers/markdown/index.js

```diff
@@ -7,9 +7,38 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import Markdown from 'markdown-it';
+import mila from 'markdown-it-link-attributes';
+import { ELASTIC_WEBSITE_URL } from '../../../public/lib/documentation_links';
+import { escapeRegExp } from '../../../common/lib/escape_reg_exp';
+
+const ELASTIC_LINK = new RegExp(
+  `^${escapeRegExp(ELASTIC_WEBSITE_URL).replace('https', 'https?')}?`
+);
+const NON_ELASTIC_EXTERNAL_LINK = new RegExp(
+  `(?!^${escapeRegExp(ELASTIC_WEBSITE_URL)})((http|https):\\/\\/[\\w\\.\\/\\-=?#]+)`
+);
 
 const md = new Markdown();
 
+md.use(mila, [
+  {
+    // forward referrer information if destination is elastic.co
+    pattern: ELASTIC_LINK,
+    attrs: {
+      target: '_blank',
```

Comment:
So... I was mistaken earlier when I said that Canvas' usage of markdown was in direction violation of https://github.com/elastic/kibana/issues/36298 because we weren't previously using `target="blank"` on these links. We still should be including "noreferrer" for third-party domains for a different reason. Do we do any "inter-Kibana" linking within Canvas, because if so, always including `target="blank"` is going to change this experience.

Reply:
You could link to other parts of Kibana using a relative URL, i.e. `kibana#/dashboard`, and it won't receive the `target="_blank"` attribute, but if you use an absolute link to another part of Kibana, it'll match my `NON_ELASTIC_EXTERNAL_LINK` regex pattern which checks that it's an absolute URL that doesn't start with `http://www.elastic.co`.

Reply:
Gotcha, that makes sense once you pointed it out...